File: ATLASDEV/AtlasModel/Enterprise/TTListPropertyDef.cs
using AtlasModel.Enterprise.Enums;
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTListPropertyDef
    {
        public Guid ListPropertyDefId { get; set; }
        public Guid ListDefId { get; set; }
        public string MemberName { get; set; }
        public string Caption { get; set; }
        public ListPropertyDefAccessEnum AccessType { get; set; }
        public byte CriteriaOrder { get; set; }
        public int ControlWidth { get; set; }
        public Guid? FormFieldDefId { get; set; }
        public short CheckOutStatus { get; set; }
    }
}

File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/DefinitionManagement/TTListPropertyDef.ts
import { Guid } from "../../Mscorlib/Guid";
import { TTObjectPropertyDef } from "./TTObjectPropertyDef";
import { ListPropertyDefAccessEnum } from "../../Utils/Enums/ListPropertyDefAccessEnum";
import { TTListDef } from "./TTListDef";
import { TTObjectCodedPropertyDef } from "./TTObjectCodedPropertyDef";
import { TTFormFieldDef } from "./TTFormFieldDef";
import { TTDataType } from "../../DataDictionary/TTDataType";

export class TTListPropertyDef {
    public ListDef: TTListDef;
    public ListDefID: Guid;
    public ListPropertyDefID: Guid;
    public PropertyDef: TTObjectPropertyDef;
    public CodedPropertyDef: TTObjectCodedPropertyDef;
    public MemberName: string;
    public DataType: TTDataType;
    public Caption: string;
    public AccessType: ListPropertyDefAccessEnum;
    public CriteriaOrder: number;
    public ControlWidth: number;
    public FormFieldDef: TTFormFieldDef;
    public FormFieldDefID: Guid;
    public IsPartialLeft: boolean;
    public IsPartialRight: boolean;
    public IsPartialFull: boolean;
    public IsPartial: boolean;
    public ToString(): string {
        return null;
    }
    public GetFilterExpression(userData: string): string {
        return null;
    }
}
